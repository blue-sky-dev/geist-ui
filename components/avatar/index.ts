import Avatar from './avatar'
import { AvatarProps } from './avatar'

export type Props = AvatarProps
export default Avatar
