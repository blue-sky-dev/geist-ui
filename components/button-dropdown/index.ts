import ButtonDropdown from './button-dropdown'
import ButtonDropdownItem from './item'

ButtonDropdown.Item = ButtonDropdownItem

export default ButtonDropdown
