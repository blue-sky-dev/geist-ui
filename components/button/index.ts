import Button from './button'
import { ButtonProps } from './button'

export type Props = ButtonProps
export default Button
