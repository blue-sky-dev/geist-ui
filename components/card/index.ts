import Card from './card'
import { CardProps } from './card'

export type Props = CardProps
export default Card
