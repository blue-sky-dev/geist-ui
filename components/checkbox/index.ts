import Checkbox from './checkbox'
import CheckboxGroup from './checkbox-group'
import { CheckboxProps, CheckboxEvent } from './checkbox'

export type Props = CheckboxProps
export type Event = CheckboxEvent

Checkbox.Group = CheckboxGroup

export default Checkbox
