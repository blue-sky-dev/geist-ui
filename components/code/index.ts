import Code from './code'
import { CodeProps } from './code'

export type Props = CodeProps
export default Code
