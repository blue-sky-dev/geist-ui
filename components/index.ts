/// <reference types="styled-jsx" />

export { default as useTheme } from './styles/use-theme'
export { default as ZEITUIProvider } from './providers/zeit-ui-provider'
export { default as useToasts } from './toast'
export { default as useInput } from './input/use-input'
export { default as useModal } from './modal/use-modal'
export { default as useTabs } from './tabs/use-tabs'
export { default as CSSBaseline } from './styles/css-baseline'
export { default as Avatar } from './avatar'
export { default as Text } from './text'
export { default as Note } from './note'
export { default as Link } from './link'
export { default as Col } from './col'
export { default as Row } from './row'
export { default as Container } from './row'
export { default as Button } from './button'
export { default as Card } from './card'
export { default as Code } from './code'
export { default as Display } from './display'
export { default as Description } from './description'
export { default as Image } from './image'
export { default as Spacer } from './spacer'
export { default as Tag } from './tag'
export { default as Dot } from './dot'
export { default as Keyboard } from './keyboard'
export { default as Checkbox } from './checkbox'
export { default as Fieldset } from './fieldset'
export { default as Modal } from './modal'
export { default as Spinner } from './spinner'
export { default as ButtonDropdown } from './button-dropdown'
export { default as Capacity } from './capacity'
export { default as Input } from './input'
export { default as Radio } from './radio'
export { default as Select } from './select'
export { default as Tabs } from './tabs'
export { default as Progress } from './progress'
export { default as Tree } from './file-tree'
export { default as Badge } from './badge'
export { default as AutoComplete } from './auto-complete'
export { default as Collapse } from './collapse'
