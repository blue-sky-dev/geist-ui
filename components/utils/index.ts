import { default as useBodyScroll } from './use-body-scroll'
import { default as useClickAnywhere } from './use-click-anywhere'
import { default as useClickAway } from './use-click-away'
import { default as useClipboard } from './use-clipboard'
import { default as useCurrentState } from './use-current-state'
import { default as useRealShape } from './use-real-shape'
import { default as useResize } from './use-resize'
import { default as useSSR } from './use-ssr'
import { default as withDefaults } from './with-defaults'

export default {
  withDefaults,
  useBodyScroll,
  useClickAnywhere,
  useClickAway,
  useClipboard,
  useCurrentState,
  useRealShape,
  useResize,
  useSSR,
}
