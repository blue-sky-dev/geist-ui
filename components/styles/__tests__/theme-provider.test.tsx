import React from 'react'
import { render, mount } from 'enzyme'
import { deepMergeObject } from '../theme-provider/theme-provider'
import DefaultThemes from '../themes/default'
import { ZeitProvider, ZeitUIThemes, Text } from 'components'
import { DeepPartial } from 'components/utils/types'

describe('ThemeProvider', () => {
  it('should deep merge objects but not add new key', () => {
    const sourceObject: any = {
      palette: {
        success: '#000',
      },
      arr: ['array', 10],
      font: {
        subFont: {
          key: 'font',
        },
      },
    }
    const targetObject: any = {
      palette: {
        warning: '#000',
        success: '#ccc',
      },
      arr: [5],
      font: {
        first: 'first',
      },
    }
    expect(deepMergeObject(sourceObject, targetObject)).toMatchObject({
      palette: {
        success: '#ccc',
      },
      arr: [5, 'array', 10],
      font: {
        subFont: {
          key: 'font',
        },
      },
    })
  })

  it('should merge themes with custom function', () => {
    const customFunc = (): DeepPartial<ZeitUIThemes> => {
      return {
        ...DefaultThemes,
        palette: {
          success: '#ccc',
        },
      }
    }
    const wrapper = render(
      <ZeitProvider theme={customFunc}>
        <Text type="success">hello</Text>
      </ZeitProvider>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should warning when using the wrong merge function', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {})
    const customFunc = () => 0 as DeepPartial<ZeitUIThemes>
    const wrapper = mount(
      <ZeitProvider theme={customFunc}>
        <p>test</p>
      </ZeitProvider>,
    )

    expect(errorSpy).toHaveBeenCalled()
    errorSpy.mockRestore()
    wrapper.unmount()
  })
})
