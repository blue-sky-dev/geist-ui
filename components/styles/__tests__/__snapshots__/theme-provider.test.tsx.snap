// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider should merge themes with custom function 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": " ",
    },
    "children": Array [
      Object {
        "data": "hello",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "p",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "data": "
        p {
          color: #ccc;
        }

        .custom-size {
          font-size: inherit;
        }
      ",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "style",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "style",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "
        p {
          color: #ccc;
        }

        .custom-size {
          font-size: inherit;
        }
      ",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "style",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "style",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "1": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "
        p {
          color: #ccc;
        }

        .custom-size {
          font-size: inherit;
        }
      ",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "style",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {
        "class": " ",
      },
      "children": Array [
        Object {
          "data": "hello",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "p",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {
            "class": " ",
          },
          "children": Array [
            Object {
              "data": "hello",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "style",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
